Mathlib/Topology/Algebra/Module/WeakDual.lean

```diff
@@ -326,6 +353,64 @@ theorem coe_map (f : E →L[𝕜] F) : (WeakSpace.map f : E → F) = f :=
   rfl
 #align weak_space.coe_map WeakSpace.coe_map
 
+/-- The canonical preimage in `E` of an open set in `WeakSpace 𝕜 E` is open in `E`. -/
+theorem isOpen_of_isOpen_WeakSpace (U : Set (WeakSpace 𝕜 E))
+    (hU : IsOpen[(WeakSpace.instTopologicalSpace : TopologicalSpace (WeakSpace 𝕜 E))] U) :
+    IsOpen ((continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' U) := by
+  exact ((continuousLinearMapToWeakSpace 𝕜 E).cont).isOpen_preimage U hU
+
+/-- A set in `E` which is open in the weak topology is open. -/
+theorem isOpen_of_isOpen_WeakSpace' (V : Set E)
+    (hV : IsOpen[(WeakSpace.instTopologicalSpace)]
+    ((continuousLinearMapToWeakSpace 𝕜 E) '' V : Set (WeakSpace 𝕜 E))) : IsOpen V := by
+  have : (continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' ((continuousLinearMapToWeakSpace 𝕜 E) '' V)
+  = V := by
+   ext x
+   constructor
+   · intro hx
+     obtain ⟨y, hy⟩ := hx
+     rw [toWeakSpace_eq_continuousLinearMapToWeakSpace,
+       toWeakSpace_eq_continuousLinearMapToWeakSpace] at hy
+     have : y = x := LinearEquiv.injective (toWeakSpace 𝕜 E) hy.2
+     rw [this] at hy
+     exact hy.1
+   · intro hx
+     simp only [Set.mem_preimage, Set.mem_image]
+     use x
+  rw [← this]
+  exact isOpen_of_isOpen_WeakSpace ((continuousLinearMapToWeakSpace 𝕜 E) '' V) hV
+
+/-- If `f : α → E` is convergent in the original topology,
+then it is convergent in the weak topology. -/
+theorem tendsto_WeakSpace_of_tendsto
+    {α : Type*} {l : Filter α} {f : α → E} {p : E} (hf : Tendsto f l (nhds p)) :
+    Tendsto (fun x ↦ (continuousLinearMapToWeakSpace 𝕜 E) (f x))
+    l (nhds ((continuousLinearMapToWeakSpace 𝕜 E) p)) := by
+  refine tendsto_nhds.mpr ?_
+  intro U hpU hU
+  rw [tendsto_nhds] at hf
+  have : (fun x ↦ (continuousLinearMapToWeakSpace 𝕜 E) (f x))
+    = (fun x ↦ ((continuousLinearMapToWeakSpace 𝕜 E) ∘ f) x) := by
+    rfl
+  rw [this, Set.preimage_comp]
+  apply hf ((continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' U)
+    ((continuousLinearMapToWeakSpace 𝕜 E).cont.isOpen_preimage U hpU)
+  exact hU
+
+/-- If `f : α → E` is convergent in the original topology,
+then `y ∘ f` is convergent for any `y : E →L[𝕜] 𝕜`. -/
+theorem eval_tendsto_of_tendsto
+    {α : Type*} {l : Filter α} {f : α → E} {p : E} (hf : Tendsto f l (nhds p)) (y : E →L[𝕜] 𝕜) :
+    Tendsto (fun x => y (f x)) l (nhds (y p)) := by
+  exact Tendsto.comp (Continuous.tendsto (ContinuousLinearMap.continuous y) p) hf
+
+/-- If `f : E → α` is continuous from `E` in the weak topology,
+then it is continuous in the original topology. -/
+theorem continuous_of_continuous_WeakSpace
+    {α : Type*} [TopologicalSpace α] {f : (WeakSpace 𝕜 E) → α} (hf : Continuous f) :
+    Continuous (f ∘ (continuousLinearMapToWeakSpace 𝕜 E)) := by
+  exact Continuous.comp hf (continuousLinearMapToWeakSpace 𝕜 E).cont
```

Comment:
I don't see why we want to have these lemmas, they don't contain anything special about `WeakSpace`, but are simply statements about continuity and convergence.

Reply:
It is true that it contains nothing special about `WeakSpace`, but as a mathematician I often say "a sequence is convergent in norm so it is convergent in the weak topology", or "a function is continuous from the weak topology, so it is continuous from the norm topology". Isn't it convenient to have such lemmas?

Reply:
Not really, the first one can be written as `(y.continuous.tendsto p).comp hf`, that is almost as short as `WeakSpace.eval_tendsto_of_tendsto  hf y`. The second one is `hf.comp (continuousLinearMapToWeakSpace k E).continuous`.

Reply:
ok, I removed them